projects/extension/src/background/index.ts

```diff
@@ -49,19 +56,40 @@ const getTime = () => {
   }${date.getSeconds()} ${date.getMilliseconds()}`
 }
 
+const maintainArr = (arr: logStructure[], incLog: logStructure): void => {
+  if (arr.length >= 1000) arr.shift()
+  arr.push(incLog)
+}
+
+// duplicate "Error" is intentional for the extreme case of level === 0
+const logLevels = ["Error", "Error", "Warn", "Info", "Debug"]
+
 const logger = (level: number, target: string, message: string) => {
-  const logLevels = ["Error", "Warn", "Info", "Debug"]
+  const incLog = {
+    time: getTime(),
+    level: logLevels[level],
```

Comment:
```suggestion
    level,
```

Reply:
level is a number. I prefer to keep the description of the error on the logs

Reply:
But it's not the job of `index.ts` to decide which words the user is going to see in the UI

Reply:
Yeah - took me few mins to "get" it - thus I removed my comment. Thank you 🙏🏽